src/zenml/orchestrators/step_launcher.py

```diff
@@ -215,13 +220,16 @@ def _create_or_reuse_run(self) -> PipelineRunResponseModel:
             stack=self._deployment.stack_id,
             pipeline=self._deployment.pipeline_id,
             enable_cache=self._deployment.pipeline.enable_cache,
+            enable_artifact_metadata=self._deployment.pipeline.enable_artifact_metadata,
             status=ExecutionStatus.RUNNING,
             pipeline_configuration=self._deployment.pipeline.dict(),
             num_steps=len(self._deployment.steps),
             client_environment=self._deployment.client_environment,
             orchestrator_environment=get_run_environment_dict(),
         )
-        return client.zen_store.get_or_create_run(pipeline_run)
+        run_response = client.zen_store.get_or_create_run(pipeline_run)
+        run_was_created = run_response.name == run_name
```

Comment:
This logic is flawed I think, what if the run name is the same, e.g. a fixed run name specified by the user?

Reply:
Mhhh yeah true. Any suggestions how this could be done cleaner?

Reply:
I added a dedicated `get_or_create_run` endpoint now that also returns a boolean of whether the run was created or not and adjusted all corresponding ZenStore methods accordingly.